neps/nep-0508.md

```diff
@@ -0,0 +1,228 @@
+---
+NEP: 508
+Title: Resharding v2
+Authors: Waclaw Banasik, Shreyan Gupta, Yoon Hong
+Status: Draft
+DiscussionsTo: https://github.com/near/nearcore/issues/8992
+Type: Protocol
+Version: 1.0.0
+Created: 2022-09-19
+LastUpdated: 2023-09-19
+---
+
+## Summary
+
+This proposal introduces a new implementation for resharding and a new shard layout for the production networks. 
+
+In essence, this NEP is an extension of [NEP-40](https://github.com/near/NEPs/blob/master/specs/Proposals/0040-split-states.md), which was focused on splitting one shard into multiple shards. 
+
+We are introducing resharding v2, which supports one shard splitting into two within one epoch at a pre-determined split boundary. The NEP includes performance improvement to make resharding feasible under the current state as well as actual resharding in mainnet and testnet (To be specific, spliting shard 3 into two). 
+
+While the new approach addresses critical limitations left unsolved in NEP-40 and is expected to remain valid for foreseable future, it does not serve all usecases, such as dynamic resharding. 
+
+## Motivation
+
+Currently, NEAR protocol has four shards. With more partners onboarding, we started seeing that some shards occasionally become over-crowded. In addition, with state sync and stateless validation, validators will not need to track all shards and validator hardware requirements can be greatly reduced with smaller shard size. 
+
+## Specification
+
+### High level assumptions
+
+* Flat state is enabled.
+* Shard split boundary is predetermined. In other words, necessity of shard splitting is manually decided.
+* Merkle Patricia Trie is the undelying data structure for the protocol state.
+* Minimal epoch gap between two resharding events is X.
+
+### High level requirements
+
+* Resharding should work even when validators stop tracking all shards.
+* Resharding should work after stateless validation is enabled.
```

Comment:
Should this be a part of the NEP? We are technically not handling stateless validation.

Reply:
we can move this to future follow ups?